### ISSUE TITLE
Multiprocessing improvement

### ISSUE THREAD

**Issue by phigas** [anufrievroman/freepaths#12]
Currently the Multiprocessing works by assigning the phonons to the subprocesses before they are started which results in some finishing before others. 
Ideally this should be handled by a worker pool which assigns the phonons to the subprocesses while they run.

**Comment by anufrievroman** [comment]
This is true, but it appears that it's only a problem for a small number of phonons. For a sufficiently large number, all workers will take roughly the same time, so that the benefit of this optimization would be rather small.
